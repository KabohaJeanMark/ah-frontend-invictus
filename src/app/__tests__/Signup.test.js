import React from 'react';
import { shallow } from "enzyme";
import SignupFormComponent from "../components/Signup";
import toJson from 'enzyme-to-json';

describe('<SignupFormComponent/>', () => {
  let wrapper;
  const PropsTest = {
    signupResponse: {},
    signupErrors: {},
    handleChange: jest.fn(),
    handleSignup: jest.fn(),
    timeMessage: jest.fn(),
    isLoading: false,
    state: {
      signup: {
        signupErrors: {},
        signupResponse: {}
      }
    }
  };
  it('renders signup page', () => {
    wrapper = shallow(<SignupFormComponent {...PropsTest} />);
  });
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

