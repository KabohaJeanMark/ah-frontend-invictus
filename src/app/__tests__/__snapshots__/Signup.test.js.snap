// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupFormComponent/> matches snapshot 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col-md-4"
    >
      <div
        className="banner"
      >
        <img
          alt="Banner"
          className="banner-img"
          src="ah_banner.png"
        />
      </div>
      <div
        className="card"
      >
        <form
          id="signup-form"
          onSubmit={[MockFunction]}
        >
          <input
            className="form-control input-field"
            id="username"
            onChange={[MockFunction]}
            placeholder="username"
            type="text"
          />
          <p />
          <input
            className="form-control input-field"
            id="email"
            onChange={[MockFunction]}
            placeholder="email"
            type="text"
          />
          <p />
          <input
            className="form-control input-field"
            id="password"
            onChange={[MockFunction]}
            placeholder="password"
            type="password"
          />
          <p />
          <button
            className="signup-btn"
            type="submit"
          >
            Signup
          </button>
          <div
            className="signupwith"
          >
            <img
              alt="signup-with"
              className="signupwith-img"
              src="signupwith.png"
            />
            <div
              className="row social-media-row"
            >
              <div
                className="col-md-1"
              />
              <div
                className="col-md-10"
              >
                <Connect(SocialAuthentication) />
              </div>
              <div
                className="col-md-1"
              />
            </div>
          </div>
          <p
            className="have-an-account"
          >
            Already have an account!
            <Link
              to="/login"
            >
              Login
            </Link>
          </p>
        </form>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</Fragment>
`;
